src/main/java/ca/nines/ise/writer/XMLWriter.java

```diff
@@ -16,179 +16,1303 @@
-          break;
-        case TEXT:
-          Text t = xml.createTextNode(n.getText());
-          xmlStack.peekFirst().appendChild(t);
-          break;
-        default:
-          throw new Exception("Cannot convert " + n.getName() + " to XML");
-      }
-    }
-
-    try {
-      Transformer transformer = TransformerFactory.newInstance().newTransformer();
-      transformer.setOutputProperty(OutputKeys.INDENT, "no");
-      transformer.setOutputProperty(OutputKeys.OMIT_XML_DECLARATION, "yes");
-      transformer.setOutputProperty(OutputKeys.ENCODING, "UTF-8");
-      DOMSource source = new DOMSource(xml);
-      StreamResult stream = new StreamResult(out);
-      transformer.transform(source, stream);
-    } catch (Exception ex) {
-      ex.printStackTrace();
-    }
-  }
-
-  /**
-   * Unsupported.
-   *
-   * @param dom
-   * @param ann
-   * @throws TransformerConfigurationException
-   * @throws TransformerException
-   * @throws IOException
-   * @throws Exception
-   */
-  @Override
-  public void render(DOM dom, Annotation ann) throws TransformerConfigurationException, TransformerException, IOException, Exception {
-    throw new UnsupportedOperationException("Not supported."); //To change body of generated methods, choose Tools | Templates.
-  }
+
+import nu.xom.Attribute;
+import nu.xom.Document;
+import nu.xom.Element;
+import nu.xom.Elements;
+import nu.xom.Text;
+import nu.xom.Serializer;
+
+public class XMLWriter extends Writer{
+	public static final String DOC_NS = "http://internetshakespeare.uvic.ca/exist/rest/db/apps/iseapp/content/schema/text/documentClass/dramaticWork.rng";
+    public static final String LINK_NS = "http://internetshakespeare.uvic.ca/#internal-linking";
+    public static final String HTML_NS = "http://www.w3.org/1999/xhtml";
+    
+    static final Map<String , String> XML_MAP = new HashMap<String , String>() {{
+    	put("RA", "right");
+    	put("C", "center");
+    	put("J", "justify");
+    	put("TITLEHEAD","title");
+    	put("VERSEQUOTE","quote");
+    }};
+    
+    /*elements in which text should not be parsed in parse_text()*/
+    public static final String[] DONT_PARSE_TEXT = {
+    	"cw",
+    	"sig",
+    	"rt",
+    	"pn",
+		"label"
+    };
+    
+    public static final String[] LINE_PARENTS = {
+    	"work",
+    	"marg",
+    	"bracegroup",
+    	"splitline",
+    	"quote",
+    	"linegroup"
+    };
+    
+    public static final String[] TYPEFACES = {
+    	"bll",
+    	"r"
+    };
+    
+    public static final String[] FLATTEN = {
+    	"font"
+    };
+	  
+    public static final String[] INLINE_TAGS = {
+    	"ABBR",
+    	"BLL",
+    	"CL",
+    	"EM",
+    	"FONT",
+    	"FOREIGN", 
+    	"HW",
+    	"I",
+    	"LD",
+    	"LS",
+    	"ORNAMENT",
+    	"PERSON",
+    	"PLACE",
+    	"PROP",
+    	"Q",
+    	"R",
+    	"S",
+    	"SC",
+    	"SD",
+    	"SUB",
+    	"SUP",
+    	"SWASH",
+    	"TITLE"
+    };
+
+	public class XMLStack extends LinkedList<Element> {
```

Comment:
It certainly can (and should) be both. I'll add the documentation. I think I missed a few of the more recent functions as well.